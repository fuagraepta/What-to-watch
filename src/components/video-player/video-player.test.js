import React from 'react';
import renderer from 'react-test-renderer';

import VideoPlayer from './video-player.jsx';

describe(`The component is rendered correctly`, () => {
  it(`Video player correctly renders, when the video is paused`, () => {
    const videoPlayer = renderer
    .create(<VideoPlayer
      renderVideo = {jest.fn()}
      onPlaybackActivate = {jest.fn()}
      onFullScreenActivate = {jest.fn()}
      isPlaying = {false}
      duration = {30}
    />)
    .toJSON();

    expect(videoPlayer).toMatchSnapshot();
  });

  it(`Video player correctly renders, when the video is played`, () => {
    const videoPlayer = renderer
      .create(<VideoPlayer
        renderVideo = {jest.fn()}
        onPlaybackActivate = {jest.fn()}
        onFullScreenActivate = {jest.fn()}
        isPlaying = {true}
        duration = {20}
      />)
      .toJSON();

    expect(videoPlayer).toMatchSnapshot();
  });
});
