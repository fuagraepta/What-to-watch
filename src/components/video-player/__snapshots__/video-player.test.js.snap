// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component is rendered correctly Video player correctly renders with transferred mock data 1`] = `
<div
  className="small-movie-card__image"
>
  <video
    style={
      Object {
        "height": "100%",
        "objectFit": "fill",
        "width": "100%",
      }
    }
  />
</div>
`;
